### PR TITLE
add smart transfer endpoints

### PR DESCRIPTION
**Pull request by dpetrovic-fb**
Add smart transfer endpoints to SDK

### CODE REVIEW

File: fireblocks_sdk/api_types.py [hunk 1]
```diff
@@ -261,6 +261,36 @@ def __init__(self, total_amount_larger_than=None, asset_id=None, order_by=None,
...
+        external_ref_id (string, optional): external ref id
+        after (string, optional): cursor string received from previous request
+        limit (number, optional): Results page size
+
+    Constraints
+        - You should only insert 'before' or 'after' (or none of them), but not both
+    """
+
+    def __init__(self, q=None, statuses=None, network_id=None, created_by_me=None, expires_after=None,
+                 expires_before=None, ticket_type=None, external_ref_id=None, after=None, limit=None):
```
Comment (idanya) [inline fireblocks_sdk/api_types.py:283]: missing types for the args

File: fireblocks_sdk/sdk.py [hunk 2]
```diff
@@ -1800,6 +1801,304 @@ def remove_web3_connection(self, session_id: str):
...
+        """Gets a page of smart transfer for your tenant according to filters given
+        Args:
+            paged_smart_transfer_request_filters (object, optional): Possible filters to apply for request
+        """
+
+        url = f"/v1/smart-transfers"
+
+        q, statuses, network_id, created_by_me, expires_after, expires_before, ticket_type, external_ref_id, limit,\
+            after = attrgetter('q', 'statuses', 'network_id', 'created_by_me', 'expires_after', 'expires_before',
+                               'ticket_type', 'external_ref_id', 'limit', 'after')(paged_smart_transfer_request_filters)
```
Comment (idanya) [inline fireblocks_sdk/sdk.py:1814]: why this method of getting attributes? 
why not just `paged_smart_transfer_request_filters.<field>` when needed?

File: fireblocks_sdk/sdk.py [hunk 3]
```diff
@@ -1800,6 +1801,304 @@ def remove_web3_connection(self, session_id: str):
...
+            terms = []
+
+        payload = {
+            "createdByNetworkId": str(created_by_network_id),
+            "type": str(ticket_type),
+            "expiresIn": int(expires_in),
+            "terms": terms,
+            "externalRefId": str(external_ref_id),
+            "note": str(note),
+            "submit": bool(submit)
```
Comment (idanya) [inline fireblocks_sdk/sdk.py:1886]: no need for casting, the type is already defined in the function signature.

Comment (dpetrovic-fb) [reply fireblocks_sdk/sdk.py]: @idanya can you please re-review this.

**Comment by dpetrovic-fb** [comment]
Hey @idanya, can you please re review this PR? Thanks

File: fireblocks_sdk/sdk.py [hunk 4]
```diff
@@ -1967,6 +1967,300 @@ def publish_policy_rules(self, rules: List[PolicyRule]):
...
+            idempotency_key (str): Idempotency key
+        """
+
+        url = f"/v1/smart-transfers/{ticket_id}/terms"
+
+        payload = {
+            "asset": str(asset),
+            "amount": amount,
+            "fromNetworkId": str(from_network_id),
+            "toNetworkId": str(to_network_id),
```
Comment (idanya) [inline fireblocks_sdk/sdk.py:2146]: casting not needed

File: fireblocks_sdk/sdk.py [hunk 5]
```diff
@@ -1967,6 +1967,300 @@ def publish_policy_rules(self, rules: List[PolicyRule]):
...
+            to_network_id (str): Destination network id
+        """
+
+        url = f"/v1/smart-transfers/{ticket_id}/terms/{term_id}"
+
+        payload = {
+            "asset": str(asset),
+            "amount": amount,
+            "fromNetworkId": str(from_network_id),
+            "toNetworkId": str(to_network_id),
```
Comment (idanya) [inline fireblocks_sdk/sdk.py:2180]: casting not needed

File: fireblocks_sdk/sdk.py [hunk 6]
```diff
@@ -1967,6 +1967,300 @@ def publish_policy_rules(self, rules: List[PolicyRule]):
...
+        url = f"/v1/smart-transfers/{ticket_id}/terms/{term_id}/fund"
+
+        payload = {
+            "asset": str(asset),
+            "amount": str(amount),
+            "networkConnectionId": str(network_connection_id),
+            "srcId": str(source_id),
+            "srcType": str(source_type),
+            "fee": str(fee),
+            "feeLevel": str(fee_level),
```
Comment (idanya) [inline fireblocks_sdk/sdk.py:2221]: casting not needed

File: fireblocks_sdk/sdk.py [hunk 7]
```diff
@@ -1967,6 +1967,300 @@ def publish_policy_rules(self, rules: List[PolicyRule]):
...
+        Args:
+            ticket_id (str): Ticket ID
+            term_id (str): Term ID
+            tx_hash (str): Transaction hash
+        """
+
+        url = f"/v1/smart-transfers/{ticket_id}/terms/{term_id}/manually-fund"
+
+        payload = {
+            "txHash": str(tx_hash),
```
Comment (idanya) [inline fireblocks_sdk/sdk.py:2237]: casting not needed

File: fireblocks_sdk/sdk.py [hunk 8]
```diff
@@ -1967,6 +1967,300 @@ def publish_policy_rules(self, rules: List[PolicyRule]):
...
+            "userGroupIds": user_group_ids,
+        }
+
+        return self._post_request(url, payload)
+
+    def get_smart_transfer_user_group_ids(self):
+        """Fetch Smart Transfer user group ids that will receive Smart Transfer notifications
+        """
+
+        url = f"/v1/smart-transfers/settings/user-groups"
```
Comment (idanya) [inline fireblocks_sdk/sdk.py:2260]: no need for f string

File: fireblocks_sdk/sdk.py [hunk 9]
```diff
@@ -1967,6 +1967,300 @@ def publish_policy_rules(self, rules: List[PolicyRule]):
...
+
+        return self._put_request(url, payload)
+
+    def set_smart_transfer_user_group_ids(self, user_group_ids):
+        """Set Smart Transfer user group ids
+        Args:
+            user_group_ids (list): List of user groups ids to receive Smart Transfer notifications
+        """
+
+        url = f"/v1/smart-transfers/settings/user-groups"
```
Comment (idanya) [inline fireblocks_sdk/sdk.py:2248]: no need for f string

File: fireblocks_sdk/sdk.py [hunk 10]
```diff
@@ -1967,6 +1967,300 @@ def publish_policy_rules(self, rules: List[PolicyRule]):
...
+        """Submit Smart Transfer ticket - change status to OPEN
+        Args:
+            ticket_id (str): ID of the ticket
+            expires_in (int): Expires in (number of hours)
+        """
+
+        url = f"/v1/smart-transfers/{ticket_id}/submit"
+
+        payload = {
+            "expiresIn": int(expires_in)
```
Comment (idanya) [inline fireblocks_sdk/sdk.py:2103]: no need to cast

File: fireblocks_sdk/sdk.py [hunk 11]
```diff
@@ -1967,6 +1967,300 @@ def publish_policy_rules(self, rules: List[PolicyRule]):
...
+        """Set External Ref. ID for Ticket
+        Args:
+            ticket_id (str): ID of the ticket
+            external_ref_id (str): ticket External Ref. id
+        """
+
+        url = f"/v1/smart-transfers/{ticket_id}/external-id"
+
+        payload = {
+            "externalRefId": str(external_ref_id)
```
Comment (idanya) [inline fireblocks_sdk/sdk.py:2088]: no need to cast

File: fireblocks_sdk/sdk.py [hunk 12]
```diff
@@ -1967,6 +1967,300 @@ def publish_policy_rules(self, rules: List[PolicyRule]):
...
+        """Set expiration for ticket.
+        Args:
+            ticket_id (str): ID of the ticket
+            expires_in (int): Expires in (number of hours)
+        """
+
+        url = f"/v1/smart-transfers/{ticket_id}/expires-in"
+
+        payload = {
+            "expiresIn": int(expires_in)
```
Comment (idanya) [inline fireblocks_sdk/sdk.py:2073]: no need to cast

File: fireblocks_sdk/api_types.py [hunk 13]
```diff
@@ -283,6 +283,39 @@ def __init__(self, total_amount_larger_than=None, asset_id=None, order_by=None,
...
+        limit (number, optional): Results page size
+
+    Constraints
+        - You should only insert 'before' or 'after' (or none of them), but not both
+    """
+
+    def __init__(self, q: str = None, statuses: str = None, network_id: str = None, created_by_me: bool = None,
+                 expires_after: str = None, expires_before: str = None, ticket_type: str = None,
+                 external_ref_id: str = None, after: str = None, limit: str = None):
+        self.q = q
```
Comment (idanya) [inline fireblocks_sdk/api_types.py:307]: rename to 'query'

File: fireblocks_sdk/api_types.py [hunk 14]
```diff
@@ -283,6 +283,39 @@ def __init__(self, total_amount_larger_than=None, asset_id=None, order_by=None,
...
+        after (string, optional): cursor string received from previous request
+        limit (number, optional): Results page size
+
+    Constraints
+        - You should only insert 'before' or 'after' (or none of them), but not both
+    """
+
+    def __init__(self, q: str = None, statuses: str = None, network_id: str = None, created_by_me: bool = None,
+                 expires_after: str = None, expires_before: str = None, ticket_type: str = None,
+                 external_ref_id: str = None, after: str = None, limit: str = None):
```
Comment (idanya) [inline fireblocks_sdk/api_types.py:306]: when you do `= None` the type can be None and not only the str/bool declared. The type should be wrapped with `Optional[]`

Comment (idanya) [reply fireblocks_sdk/api_types.py:306]: Please update all functions (not only this one)

Comment (dpetrovic-fb) [reply fireblocks_sdk/api_types.py]: done. Can you please re-review?

File: fireblocks_sdk/sdk.py (hunk 12 above)
Comment (dpetrovic-fb) [reply fireblocks_sdk/sdk.py]: done. Can you please re-review?

**Review by idanya — APPROVED** [review]

**Pull request merged**